urlextract/urlextract_core.py

```diff
@@ -115,6 +115,11 @@ def __init__(
         # characters that are allowed to be right after TLD
         self._after_tld_chars = self._get_after_tld_chars()
 
+        # protocol char
+        self._stop_protocol_char = ":"
+        # TODO: possibly add "ftp" and more
+        self._supported_protocols = ["http", "https", "file"]
```

Comment:
This might reduce found URLs by significant amount. 
See registered schemes: https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml

Reply:
Nitpick: in URL world `http` is a schema (not protocol) ;-)

Reply:
I see your point, ok, i guess the workaround would be to open stop character (or better - "term") expansion such that as a consumer of this I can specify a custom set of stop terms. Currently you have it implemented as characters but it's not difficult to flip it around to use terms.

Reply:
Also you probably do not want to include the IANA schemas as it will slow down your code. Already as is, running this on a Pandas set of 56K records  of social network posts takes a WHILE - about a minute (vs say compiled Regex which takes a few seconds). So improving performance here would be key imo.